test/utils/cluster.go

```diff
@@ -0,0 +1,123 @@
+// Copyright 2023 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package utils
+
+import (
+	"fmt"
+	"os/exec"
+	"strconv"
+	"strings"
+
+	"k8s.io/klog/v2"
+)
+
+const listFormat = "value(NAME)"
+
+func EnsureCluster(clusterName, zone string, numOfNodes int) error {
+	if !isClusterExisting(clusterName, zone) {
+		klog.Infof("Cluster %s in zone %s does not exist, creating.", clusterName, zone)
+		return createCluster(clusterName, zone, numOfNodes)
+	}
+	params := []string{
+		"container",
+		"clusters",
+		"describe",
+		clusterName,
+		"--zone", zone,
+	}
+	out, err := exec.Command("gcloud", params...).CombinedOutput()
+	if err != nil {
+		return fmt.Errorf("cannot describe cluster %s using gcloud: %w", clusterName, err)
+	}
+	if err := verifyClusterInfo(string(out), numOfNodes); err != nil {
```

Comment:
move params, exec.Command into verifyCluster(name, clusterConfig)

this will keep the pieces together inside that func

Reply:
Updated.